src/test/java/com/commercetools/sync/cartdiscounts/CartDiscountSyncOptionsBuilderTest.java

```diff
@@ -0,0 +1,273 @@
+        final List<UpdateAction<CartDiscount>> filteredList =
+            cartDiscountSyncOptions
+                .applyBeforeUpdateCallBack(updateActions, mock(CartDiscountDraft.class), mock(CartDiscount.class));
+        assertThat(filteredList).isNotEqualTo(updateActions);
+        assertThat(filteredList).isEmpty();
+    }
+
+    private interface MockTriFunction extends
+        TriFunction<List<UpdateAction<CartDiscount>>, CartDiscountDraft,
+            CartDiscount, List<UpdateAction<CartDiscount>>> {
+    }
+
+    @Test
+    public void applyBeforeUpdateCallBack_WithEmptyUpdateActions_ShouldNotApplyBeforeUpdateCallback() {
+        final MockTriFunction beforeUpdateCallback = mock(MockTriFunction.class);
+
+        final CartDiscountSyncOptions cartDiscountSyncOptions =
+            CartDiscountSyncOptionsBuilder.of(CTP_CLIENT)
+                                          .beforeUpdateCallback(beforeUpdateCallback)
+                                          .build();
+
+        assertThat(cartDiscountSyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        final List<UpdateAction<CartDiscount>> updateActions = emptyList();
+        final List<UpdateAction<CartDiscount>> filteredList =
+            cartDiscountSyncOptions
+                .applyBeforeUpdateCallBack(updateActions, mock(CartDiscountDraft.class), mock(CartDiscount.class));
+
+        assertThat(filteredList).isEmpty();
+        verify(beforeUpdateCallback, never()).apply(any(), any(), any());
+    }
+
+    @Test
+    public void applyBeforeUpdateCallBack_WithCallback_ShouldReturnFilteredList() {
+        final TriFunction<List<UpdateAction<CartDiscount>>, CartDiscountDraft,
+            CartDiscount, List<UpdateAction<CartDiscount>>> beforeUpdateCallback =
+                (updateActions, newCartDiscount, oldCartDiscount) -> emptyList();
+
+        final CartDiscountSyncOptions cartDiscountSyncOptions = CartDiscountSyncOptionsBuilder.of(CTP_CLIENT)
+                                                                                      .beforeUpdateCallback(
+                                                                                          beforeUpdateCallback)
+                                                                                      .build();
+        assertThat(cartDiscountSyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        final List<UpdateAction<CartDiscount>> updateActions = singletonList(ChangeName.of(ofEnglish("name")));
+        final List<UpdateAction<CartDiscount>> filteredList =
+            cartDiscountSyncOptions
+                .applyBeforeUpdateCallBack(updateActions, mock(CartDiscountDraft.class), mock(CartDiscount.class));
+        assertThat(filteredList).isNotEqualTo(updateActions);
+        assertThat(filteredList).isEmpty();
+    }
+
+    @Test
+    public void applyBeforeCreateCallBack_WithCallback_ShouldReturnFilteredDraft() {
+        //todo: there is no key, instead of key, use name as an option
```

Comment:
But actually why do we need this TODO? We don't need to use the key here at all? It's just a test for the callback, so the key/name of the draft doesn't matter which is used.

Reply:
you are goddamn right 😄